### ISSUE TITLE
Cleanup duplicated interop calls in PlatformDetection to use ones from src/common instead

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/corefx#32438]
A lot of interop calls are duplicated. Include common files instead: https://github.com/dotnet/corefx/pull/32433#discussion_r219922828

**Comment by artkpv** [comment]
Hello, I'm removing these duplicates. Did for `RTL_OSVERSIONINFOEX` and stopped at `OpenProcessToken`. Didn't find for `GetCurrentApplicationUserModelId` and `CloseHandle`. Probably to add into `src\Common`?
Some progress at https://github.com/dotnet/corefx/compare/master...artkpv:duplicates-32438

**Comment by danmosemsft** [comment]
@artkpv I sent you a collaborator link. If you accept, I can assign this issue to you. Note: being a collaborator auto subscribes to the repo, you may want to adjust that after accepting.

**Comment by artkpv** [comment]
Reporting the progress. Now I've stopped at getting `SafeTokenHandle` to use `OpenProcessToken(..)`. The difficulty is that I don't understand why if I reference `System.Diagnostics.Process` lib with this class my `CoreFx.Private.TestUtilities` can not find it (`error CS0246: The type or namespace name 'SafeTokenHandle' could not be found`). It uses older lib in `.\bin\ref\System.Diagnostics.Process\4.2.1.0\..`. So I'm reading ([this](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md) and [that](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md)) and experimenting on how to reference projects. Looks tangled so far :)

**Comment by ViktorHofer** [comment]
If you push your changes I can take a look.

**Comment by ViktorHofer** [comment]
Hadn't had time to look into it. Maybe somebody else can help unblocking @artkpv.

cc @anipik @danmosemsft)

**Comment by Anipik** [comment]
@artkpv you are getting this error because ```SafeTokenHandle``` is an internal class . which ```OpenProcessToken``` you are trying to remove ?

**Comment by artkpv** [comment]
@Anipik  Yes, missed that. This is to reuse `Common\src\Interop\Windows\advapi32\Interop.OpenProcessToken.cs` in `CoreFx.Private.TestUtilities`.  That's inconsistency, I image, as external method in Common uses class which is not there. So I guess solution would be to move `SafeTokenHandle` into `Common\src\Microsoft\Win32\SafeHandles\` and use it in 3 libs (`System.Diagnostics.Process`, `System.Security.AccessControl` and the one).

**Comment by Anipik** [comment]
I dont see any difficulty\problem in combining SafeTokenHandle class from https://github.com/dotnet/corefx/blob/d2b05edb02129eb8aad4331feb7e5b920afbb4c5/src/System.Diagnostics.Process/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs and https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Security.AccessControl/src/System/Security/SafeSecurityHandles.cs
You can go ahead and do it

cc @danmosemsft

**Comment by artkpv** [comment]
@Anipik It is unfinished from #5209 times. 

After looking at it more, the same is here and there. Probably should be ticket for this. Below is the list of files with the same class as I see.

1) `public` "System.Security.Principal.Windows\src\Microsoft\Win32\SafeHandles\SafeAccessTokenHandle.cs" 
[Doc](https://docs.microsoft.com/en-us/dotnet/api/microsoft.win32.safehandles.safeaccesstokenhandle?view=netcore-2.1): "Provides a safe handle to a Windows thread or process access token." 
Used in 
CoreFx.Private.TestUtilities (3)
System.Diagnostics.PerformanceCounter (1)
System.IO.Pipes.Tests (2)
System.Net.Security (2)
System.Security.Principal.Windows (41)
System.Security.Principal.Windows.Tests (4)

2) `internal` "System.Security.AccessControl/src/System/Security/SafeSecurityHandles.cs"
In System.Security.AccessControl (14)

3) `internal` "Common\src\Microsoft\Win32\SafeHandles\SafeTokenHandle.cs" 
In System.Diagnostics.Process (5)

4) `public partial calss SafeProcessHandle : SafeHandleZeroOrMinusOneIsInvalid` System.Diagnostics.Process\src\Microsoft\Win32\SafeHandles\SafeProcessHandle.Windows.cs 
[Doc](https://docs.microsoft.com/en-us/dotnet/api/microsoft.win32.safehandles.safeprocesshandle?view=netcore-2.1): "Provides a managed wrapper for a process handle." 
In System.Diagnostics.PerformanceCounter (12)
System.Diagnostics.Process (65)
System.Diagnostics.Process.Tests (7)


5) `internal` "System.Diagnostics.Process\src\Microsoft\Win32\SafeHandles\SafeThreadHandle.cs"
Only in System.Diagnostics.Process (19).

And 8 versions for `OpenProcessToken` (https://source.dot.net/#q=OpenProcessToken)

So where what should we use? :) Probably we can use only `SafeProcessHandle` and `SafeAccessTokenHandle` as they are `public`. I'm going to try to do this next.

**Comment by artkpv** [comment]
@ViktorHofer I guess this issue can be closed. I'm going to make another issue for the above comment.
